pulsar-broker/src/main/java/org/apache/pulsar/broker/loadbalance/extensions/data/BrokerLoadData.java

```diff
@@ -0,0 +1,188 @@
+                                           final ResourceUsage directMemory, final ResourceUsage bandwidthIn,
+                                           final ResourceUsage bandwidthOut) {
+        this.cpu = cpu;
+        this.memory = memory;
+        this.directMemory = directMemory;
+        this.bandwidthIn = bandwidthIn;
+        this.bandwidthOut = bandwidthOut;
+    }
+
+    public double getMaxResourceUsage() {
+        return max(cpu.percentUsage(), directMemory.percentUsage(), bandwidthIn.percentUsage(),
+                bandwidthOut.percentUsage()) / 100;
+    }
+
+    public double getMaxResourceUsageWithWeight(final double cpuWeight, final double memoryWeight,
+                                                final double directMemoryWeight, final double bandwidthInWeight,
+                                                final double bandwidthOutWeight) {
+        return max(cpu.percentUsage() * cpuWeight, memory.percentUsage() * memoryWeight,
+                directMemory.percentUsage() * directMemoryWeight, bandwidthIn.percentUsage() * bandwidthInWeight,
+                bandwidthOut.percentUsage() * bandwidthOutWeight) / 100;
+    }
+
+    public double getMaxResourceUsageWithWeightWithinLimit(final double cpuWeight, final double memoryWeight,
+                                                           final double directMemoryWeight,
+                                                           final double bandwidthInWeight,
+                                                           final double bandwidthOutWeight) {
+        return maxWithinLimit(100.0d,
+                cpu.percentUsage() * cpuWeight, memory.percentUsage() * memoryWeight,
+                directMemory.percentUsage() * directMemoryWeight, bandwidthIn.percentUsage() * bandwidthInWeight,
+                bandwidthOut.percentUsage() * bandwidthOutWeight) / 100;
+    }
+
+    private static double getNicSpeedBytesInSec(ServiceConfiguration conf) {
+        return conf.getLoadBalancerOverrideBrokerNicSpeedGbps().isPresent()
+                ? conf.getLoadBalancerOverrideBrokerNicSpeedGbps().get() * gigaBitToByte : -1.0;
+    }
+
+    synchronized ResourceUsage getMsgThroughputInUsage(double nicSpeedBytesInSec) {
+        if (msgThroughputInUsage.usage != msgThroughputIn) {
+            msgThroughputInUsage = new ResourceUsage(msgThroughputIn, nicSpeedBytesInSec);
+        }
+        return msgThroughputInUsage;
+    }
+
+    synchronized ResourceUsage getMsgThroughputOutUsage(double nicSpeedBytesInSec) {
+        if (msgThroughputOutUsage.usage != msgThroughputOut) {
+            msgThroughputOutUsage = new ResourceUsage(msgThroughputOut, nicSpeedBytesInSec);
+        }
+        return msgThroughputOutUsage;
+    }
+
+    public double getMaxResourceUsageWithExtendedNetworkSignal(ServiceConfiguration conf) {
+
+        double nicSpeedBytesInSec = getNicSpeedBytesInSec(conf);
+        return maxWithinLimit(100.0d,
```

Comment:
And we can't apply the limitation of resource usage. 
https://github.com/apache/pulsar/issues/18598 has provided more details.